org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/LRUCache.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.ls.debug.adapter;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+public class LRUCache<K, V> implements ICache<K, V> {
```

Comment:
suggest to use `LRUCache<K, V> extends LinkedHashMap<K,V> implements ICache<K,V>`